tests/scripts/thread-cert/Cert_5_3_04_AddressMapCache.py

```diff
@@ -84,56 +91,58 @@ def setUp(self):
         self.nodes[ED4].add_whitelist(self.nodes[LEADER].get_addr64())
         self.nodes[ED4].enable_whitelist()
 
-        self.nodes[ED5].set_panid(0xface)
-        self.nodes[ED5].set_mode('rsn')
-        self.nodes[ED5].add_whitelist(self.nodes[LEADER].get_addr64())
-        self.nodes[ED5].enable_whitelist()
+        self.sniffer = config.create_default_thread_sniffer()
+        self.sniffer.start()
 
     def tearDown(self):
+        self.sniffer.stop()
+        del self.sniffer
+
         for node in list(self.nodes.values()):
             node.stop()
         del self.nodes
 
     def test(self):
+        # 1
         self.nodes[LEADER].start()
         self.nodes[LEADER].set_state('leader')
         self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
 
-        self.nodes[ROUTER1].start()
+        for i in range(2,8):
+            self.nodes[i].start()
         time.sleep(5)
-        self.assertEqual(self.nodes[ROUTER1].get_state(), 'router')
 
-        self.nodes[SED1].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[SED1].get_state(), 'child')
+        self.assertEqual(self.nodes[DUT_ROUTER1].get_state(), 'router')
 
-        self.nodes[ED2].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED2].get_state(), 'child')
+        for i in MTDS:
+            self.assertEqual(self.nodes[i].get_state(), 'child')
 
-        self.nodes[ED3].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED3].get_state(), 'child')
+        # This method flushes the message queue so calling this method again will return only the oldly logged messages.
+        dut_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER1)
 
-        self.nodes[ED4].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED4].get_state(), 'child')
+        # 2
+        for i in range(4, 8):
```

Comment:
avoid to use magic number (4, 8) if possible? e.g. here uses list [ED1, ED2, ED3, ED4] for better readability?